feat(DMS): add datasource smart connect tasks (#3864)
huaweicloud/terraform-provider-huaweicloud: docs/data-sources/dms_kafka_smart_connect_tasks.md

---
subcategory: "Distributed Message Service (DMS)"
---

# huaweicloud_dms_kafka_smart_connect_tasks

> Use this data source to get the list of DMS kafka smart connect task.

Reviewer: ```suggestion
Use this data source to get the list of DMS kafka smart connect tasks.
```

Author: done

## Example Usage

```hcl
var "connector_id" {}

data "huaweicloud_dms_kafka_smart_connect_tasks" "test" {
  connector_id     = var.connector_id

>   task_id          = "60070a02-27c7-4eee-985d-95fc67f32d85"

Reviewer: shoud use variable like connector_id

Author: done

  task_name        = "test_task"
  destination_type = "OBS"
  status           = "RUNNING"
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used.

* `connector_id` - (Required, String) Specifies the connector ID of the kafka instance.

> * `task_id` - (Optional, String) Specifies the id of the smart connect task.

Reviewer: ```suggestion
* `task_id` - (Optional, String) Specifies the ID of the smart connect task.
```

Author: done

* `task_name` - (Optional, String) Specifies the name of the smart connect task.

* `destination_type` - (Optional, String) Specifies the destination type of the smart connect task.

* `status` - (Optional, String) Specifies the status of the smart connect task. Value options: **RUNNING**, **PAUSED**.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The data source ID.

> * `tasks` - The list of smart connect task.

Reviewer: ```suggestion
* `tasks` - The list of smart connect tasks.
```

Author: done

  The [tasks](#DMS_kafka_smart_connect_tasks) structure is documented below.

<a name="DMS_kafka_smart_connect_tasks"></a>
The `tasks` block supports:

> * `task_id` - Indicates the id of the smart connect task.

Reviewer: ```suggestion
* `id` - Indicates the ID of the smart connect task.
```
and fix other places

Author: done. change task_id as id in the list of tasks.

* `task_name` - Indicates the name of the smart connect task.

* `destination_type` - Indicates the destination type of the smart connect task.

* `created_at` - Indicates the creation time of the smart connect task.

* `status` - Indicates the status of the smart connect task. The value can be: **RUNNING**, **PAUSED**.

> * `topics` - Indicates the topic names separated by a comma or the topic regular expression of the smart connect task.

Reviewer: ```suggestion
* `topics` - Indicates the topic names separated by commas or the topic regular expression of the smart connect task.
```

Author: done
